src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -126,118 +126,6 @@ namespace {
-              sp2.getStimulusThreshold());
-    ASSERT_TRUE(sp1.getDutyCyclePeriod() == sp2.getDutyCyclePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getBoostStrength(), sp2.getBoostStrength()));
-    ASSERT_TRUE(sp1.getIterationNum() == sp2.getIterationNum());
-    ASSERT_TRUE(sp1.getIterationLearnNum() ==
-              sp2.getIterationLearnNum());
-    ASSERT_TRUE(sp1.getSpVerbosity() == sp2.getSpVerbosity());
-    ASSERT_TRUE(sp1.getWrapAround() == sp2.getWrapAround());
-    ASSERT_TRUE(sp1.getUpdatePeriod() == sp2.getUpdatePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getSynPermTrimThreshold(),
-              sp2.getSynPermTrimThreshold()));
-    cout << "check: " << sp1.getSynPermActiveInc() << " " <<
-      sp2.getSynPermActiveInc() << endl;
-    ASSERT_TRUE(almost_eq(sp1.getSynPermActiveInc(),
-              sp2.getSynPermActiveInc()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermInactiveDec(),
-              sp2.getSynPermInactiveDec()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermBelowStimulusInc(),
-              sp2.getSynPermBelowStimulusInc()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermConnected(),
-              sp2.getSynPermConnected()));
-    ASSERT_TRUE(almost_eq(sp1.getMinPctOverlapDutyCycles(),
-              sp2.getMinPctOverlapDutyCycles()));
-
-
-    auto boostFactors1 = new Real[numColumns];
-    auto boostFactors2 = new Real[numColumns];
-    sp1.getBoostFactors(boostFactors1);
-    sp2.getBoostFactors(boostFactors2);
-    ASSERT_TRUE(check_vector_eq(boostFactors1, boostFactors2, numColumns));
-    delete[] boostFactors1;
-    delete[] boostFactors2;
-
-    auto overlapDutyCycles1 = new Real[numColumns];
-    auto overlapDutyCycles2 = new Real[numColumns];
-    sp1.getOverlapDutyCycles(overlapDutyCycles1);
-    sp2.getOverlapDutyCycles(overlapDutyCycles2);
-    ASSERT_TRUE(check_vector_eq(overlapDutyCycles1, overlapDutyCycles2, numColumns));
-    delete[] overlapDutyCycles1;
-    delete[] overlapDutyCycles2;
-
-    auto activeDutyCycles1 = new Real[numColumns];
-    auto activeDutyCycles2 = new Real[numColumns];
-    sp1.getActiveDutyCycles(activeDutyCycles1);
-    sp2.getActiveDutyCycles(activeDutyCycles2);
-    ASSERT_TRUE(check_vector_eq(activeDutyCycles1, activeDutyCycles2, numColumns));
-    delete[] activeDutyCycles1;
-    delete[] activeDutyCycles2;
-
-    auto minOverlapDutyCycles1 = new Real[numColumns];
-    auto minOverlapDutyCycles2 = new Real[numColumns];
-    sp1.getMinOverlapDutyCycles(minOverlapDutyCycles1);
-    sp2.getMinOverlapDutyCycles(minOverlapDutyCycles2);
-    ASSERT_TRUE(check_vector_eq(minOverlapDutyCycles1, minOverlapDutyCycles2, numColumns));
-    delete[] minOverlapDutyCycles1;
-    delete[] minOverlapDutyCycles2;
-
-    for (UInt i = 0; i < numColumns; i++) {
-      auto potential1 = new UInt[numInputs];
-      auto potential2 = new UInt[numInputs];
-      sp1.getPotential(i, potential1);
-      sp2.getPotential(i, potential2);
-      ASSERT_TRUE(check_vector_eq(potential1, potential2, numInputs));
```

Comment:
This is the case where check is within a for loop, so we'll have to create a variable that covers the whole loop: 
```
bool allPotentialsEq = true;
for (..)
  allPotentialsEq == allPotEq && check_vector_eq(..)
```

Reply:
check_vector_eq should be implemented as `std::equal()` with the "twice" example: 
http://www.tenouk.com/cpluscodesnippet/cplusstlvectoralgorithmequal.html